### DISCUSSION TITLE
Problema na conversão de moedas. [Câmbio]

### DISCUSSION THREAD

**Discussion by viniciuscsilveira** [OCA/l10n-brazil#2722] [Q&A]
Boa tarde pessoal.
Caso possam me ajudar .
Abri um chamado no suporte Odoo mas até agora nada.

Quais são as etapas para reproduzir o problema?
Bom dia. Como vendemos para o exterior e recebemos em USD, através de uma corretora. Criei uma conta corrente com o nome da corretora como se fosse um banco, pois o valor em USD permanece por um tempo com eles . No momento do câmbio que se dá pela transferência da conta da corretora em USD para a conta da empresa aqui no Brasil em real.

Qual é o comportamento atual que você observa?
Eu fiz este teste e ocorreu que no Odoo. A cotação USD x BRL fica diferente do valor da corretora dando assim. divergência de valores. Também ao fazer o processo onde preciso inserir uma negociação do dia 07/02/2023 por exemplo; 1 - Crio a cotação em USD; 2 - Crio a fatura; 3 - Executo o pagamento com a data real que foi dia 07/02/2023; Aí já acontece um problema . O valor em USD 18.580,80 muda para BRL e ao invés de fazer o câmbio não faz e em BRL registra como R$ 18.580,80. Percebi que isso acontece quando muda a data para uma data passada de 07/02/2023. Eu preciso fazer estes lançamentos do passado par evidenciar evidenciar as operações financeira , estoque e etc...

Qual seria o comportamento esperado neste caso?
A conversão de moeda acontecer de forma onde eu poderia colocar o valor da cotação referente ao dia do câmbio.
Lembro que estou inserindo informações do passado no sistema.
|
No link abaixo o vídeo mostrando o fluxo.
Desta vez. Com a data do movimento real que é dia 07/02/2023.
Veja que o valor é de 18.580,80 USD quando é convertido para BRL aparece como 18.580,80 BRL e isto está errado.
Percebi que isso acontece quando coloco a data do passado.
Vídeo: https://drive.google.com/file/d/1PK0U5AFzu_hNQTMd5arZ77Yg8HnEwxI0/view?usp=sharing
|
No link abaixo o vídeo mostrando o fluxo.
Mas a data do movimento que estou gerando na realidade é dia 07/02/2023.
Eu deixando a data de hoje consigo fazer todo fluxo, porém o câmbio no final para conversão deveria ser USD x BRL a 5,0732 BRL por USD.
Veja que o valor é de 18.580,80 USD que convertido para BRL deveria ser 94.264,1146.
Vídeo: https://drive.google.com/file/d/11zOOgWxUnD2VUpG9DibQERgGI519vEZZ/view?usp=sharing

**Comment by viniciuscsilveira** [comment]
Consegui resolver.

**Reply by douglascstd** [reply]
Compartilha aí.... hehheh

**Reply by viniciuscsilveira** [reply]
Desabilitei o cálculo automático de moedas.
E na moeda USD adicionei uma linha com o valor do câmbio que negociei junto a corretora.

**Reply by viniciuscsilveira** [reply]
https://drive.google.com/file/d/1eD3smeBfWWhQYNIaOwsGr84gZgeHBr0C/view?usp=sharing
